### ISSUE TITLE
Audit GM2 implementation vs design

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#6328]
## Feature Description

It recently came up that there are some minor differences in our GM2+ implementation vs the intended design. As part of this, we should review our design tokens to make sure we're consistent with our internal representations.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All Site Kit design tokens (i.e. CSS custom properties) should be consistent their respective definitions in the [GM2+ design](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B)
* Storybook stories should be created to cover the following panels from the design (these can be basic and can essentially mirror the layout from the design):
    * Type Scale
    * Tokens
* The added stories should have VRT scenarios enabled for them so that changes in the future will cause VRT to fail and require approval
    * If possible, we can limit the scenarios here to be desktop-only unless there is a benefit to using the default 3 viewports / scenario

## Implementation Brief

* Compare the type scale values from [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0) with the their coresponding `$fs-*` and `$lh-*` variables in [`_variables-mui3.scss`](https://github.com/google/site-kit-wp/blob/eacac2cd48ceb4cb8a72b6b6ad9a438849f73226/assets/sass/config/_variables-mui3.scss#L188). If different, adjust the `scss` values to match figma.
* Compare the token values (colors) from [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0) with the their coresponding `$c-*` variables in [`_variables-mui3.scss`](https://github.com/google/site-kit-wp/blob/eacac2cd48ceb4cb8a72b6b6ad9a438849f73226/assets/sass/config/_variables-mui3.scss#L188). If different, adjust the `scss` values to match figma.
* Add a new story `stories/type-scale.stories.js` that shows all type scales as they are shown in [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0). Replicate the "Type Scale" section from [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0). Enable VRT for this new story (see [Storybook v6 usage](https://docs.google.com/document/d/1Ny2Tlo1_FxZY6s6dC2-E1TWJYi1rhp0OMEtEfXxwXjY/edit?resourcekey=0-PFLAM17zQVq9Lb05pYlj7A#heading=h.qgpzyv3ba0sc)).
* Add a new story `stories/tokens.stories.js` that shows all tokens (colors) as they are shown in [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0). Replicate the "Tokens" section from [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=398%3A250&t=tgh6YOiihrECBn1L-0). Enable VRT for this new story (see [Storybook v6 usage](https://docs.google.com/document/d/1Ny2Tlo1_FxZY6s6dC2-E1TWJYi1rhp0OMEtEfXxwXjY/edit?resourcekey=0-PFLAM17zQVq9Lb05pYlj7A#heading=h.qgpzyv3ba0sc)).

### Test Coverage

* VRT images must be updated.

## QA Brief

* Verify the storybook story "Global > Tokens" matches the Figma color palette.
* Verify the storybook story "Global > TypeScale" matches the Figma type scale.
* Verify the updated VRT and their corresponding stories match the Figma designs.
  * `font-sizes`, `line-heights` and color tokens have been changed.
* Review Sitekit Dashboard and Settings and verify all UI elements work as expected as this issue introduces small changes to almost all views.

## Changelog entry

* Update type scale and color tokens to match GM2+ design.

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by derweili** [comment]
While working on execution, I found some inconsistencies I do not know how to resolve:

### 1. `$c-content-link`
https://github.com/google/site-kit-wp/blob/878d7fd21a61b6a1110d4dd080a8d7ce2de5a5ec/assets/sass/config/_variables-mui3.scss#L169-L170

This variables exists in our code, but do not have a matching definition in figma.
I found [this slack thread](https://10up.slack.com/archives/CFFRMC5DE/p1657700949451349), which references this variable to match `teal-500`. So I updated the definition to `$c-content-link: $c-teal-t-500;`.

But when comparing the HEX code, I found out the `$c-content-link` has the same color as `content/secondary` which by now did not exist in our code. So I'm wondering if we should update all occurences of `$c-content-link` to the new `$c-content-secondary`? Does this make sense?


### 2. `$c-content-primary-hover`
https://github.com/google/site-kit-wp/blob/878d7fd21a61b6a1110d4dd080a8d7ce2de5a5ec/assets/sass/config/_variables-mui3.scss#L171
For the `$c-content-primary-hover` I did not find any reference in figma. Just [this slack thread](https://10up.slack.com/archives/C01CA828ZPZ/p1655381794839489), where Seba posted the HEX code for this variable. But without a relation to figma. This HEX code is not used in figma for any other variable.

Do we have an official definition for the `$c-content-primary-hover` variable beside the slack messages? 

### 3. neutral schema.
In the [figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?node-id=301%3A209&t=VpYMtZnbHJsgQCqx-0) design, the `n-50` color seams to be replaced by `n-33` and `n-66`, which both do not yet exist in our code. The `*-33` and `*-66` definitions do not exist for other color schemas.

Because the `n-50` is already in use for some components, I'm unsure on how to adapt the design. Should the `n-50` be replaced? If yes, which color should be used, `n-33` or `n-66`?
Also the `utility/divider` color references the replaced `n-50`, so I'm wondering what to do with this one.

**Comment by bethanylang** [comment]
@sebastianmantel Tagging you in here per our discussion.  :)

**Comment by derweili** [comment]
@sebastianmantel
Our code includes a `$c-interactive-on-disable-container` color token, which does not exist in Figma (anymore). But there is a new `interactive/on-disable-container`. Was the colour simply renamed?
This color is currently only used for color of the user input question numbers.
https://github.com/google/site-kit-wp/blob/e577647e33e99739ddbf6e15501d55f83bd22e2f/assets/sass/components/user-input/googlesitekit-user-input-questions.scss#L52-L63

**Comment by derweili** [comment]
This issue is blocked by some questions raised in figma comments. See https://10up.slack.com/archives/C01CA828ZPZ/p1673620713885709

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@nfmohit I have a few observations to highlight.

**Storybook story `Global > Tokens`**

- Under Interactive: in FIgma the Tertiary Hover is `8%` but on Storybook it is `36%`

**Storybook story `Global > TypeScale`** 

- Under Title in Figma the font style is `Google Sans Text` but on Storybook it is `Google Sans Display` 
- Under Link in Figma the font colour is `#108080` but on Storybook it is `#161b18` and also the link is not underlined

I am wondering if this should be QA:Eng with the reference to VRT.  Is that correct?

> Verify the updated VRT and their corresponding stories match the Figma designs. font-sizes, line-heights and color tokens have been changed.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- The storybook story "Global > Tokens" matches the Figma color palette.
- The storybook story "Global > TypeScale" matches the Figma type scale.
   - font-sizes, line-heights and color tokens have been changed.

- Reviewed Sitekit Dashboard and Settings and verify all UI elements work as expected.
- The issues that I reported [here](https://github.com/google/site-kit-wp/issues/6328#issuecomment-1402095706) are fixed. 

> Verify the updated VRT and their corresponding stories match the Figma designs.

I think the VRTs is a `QA:Eng` task, so I have added the label on to the ticket for an engineer to look over.

**Comment by aaemnnosttv** [comment]
## ⚠️ Approval

I noticed a regression which seems to have been introduced here where the text color in modal dialogs is now green 😄 

![image](https://user-images.githubusercontent.com/1621608/215182187-d01a2c0e-b41a-4531-b552-b584c51caa5f.png)

See https://google.github.io/site-kit-wp/storybook/main/?path=/story/global--modal-dialog

This could be because the wrong token was used there in the past, I'm not sure, but the responsible style seems to be several months old https://github.com/google/site-kit-wp/blob/ffc74b1a62307785694b1a5b62c7026659ec6975/assets/sass/vendor/_mdc-dialog.scss#L9-L11

This is captured in the [VRT difference in the original PR](https://github.com/google/site-kit-wp/pull/6344/files#diff-cd55264739472f734b7203d86da82f89dabe6b3d074efa8f943276e487f82d6b) so let's review those differences as well to make sure there isn't anything else potentially missed here. I realize a lot of things were affected by this so it could have been easy to miss there.

**Comment by sebastianmantel** [comment]
@aaemnnosttv As I can see, Sigal made the updates on the tokens and styles that we discussed a while ago. Do we have any open issues related to GM2+ design?

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- Retested the issue that Evan has reported about the modal when disconnecting Site Kit or a module.  The correct color is now appearing, it is now #161B18.

![image](https://user-images.githubusercontent.com/73545194/215496796-1d0eb349-3507-4a3d-94ae-372b31264481.png)
